File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.py
N = int(input())

for i in range(N):
    a = int(input())
    dp = [0] * (a+1)
    for j in range(1,a+1):
        if j == 1:
            dp[j] = 1
        elif j == 2:
            dp[j] = 2
        elif j == 3:
            dp[j] = 4
        else:
            dp[j] = dp[j-1] + dp[j-2] + dp[j-3]
    print(dp[a])

File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let N = 0;
let inputLine = 0;
let answer = [];
rl.on("line", (line) => {
  if (inputLine == 0) {
    N = parseInt(line);
  }
  if (inputLine > 0 && inputLine <= N) {
    let temp = parseInt(line);
    let dp = Array.from({ length: temp + 1 }, () => 0);
    dp[1] = 1;
    dp[2] = 2;
    dp[3] = 4; //1 1 1, 1 2, 2 1, 3

    for (let i = 4; i <= temp; i++) {
      dp[i] = dp[i - 1] + dp[i - 2] + dp[i - 3];
    }
    answer.push(dp[temp]);
  }
  if (inputLine == N) {
    console.log(answer.join("\n"));
    process.exit();
  }
  inputLine++;
});
